engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -2091,8 +2091,11 @@ public void execute(String[] args, ConsoleReader reader) throws Exception
           if (!commandLineInfo.localMode) {
 
             // see whether there is an app with the same name and user name running
-            String appNameAttributeName = StreamingApplication.DT_PREFIX + Context.DAGContext.APPLICATION_NAME.getName();
-            String appName = config.get(appNameAttributeName, appFactory.getName());
+            String appNameAttributeName = StreamingApplication.APEX_PREFIX + Context.DAGContext.APPLICATION_NAME.getName();
+            String appName = config.get(appNameAttributeName);
```

Comment:
addDeprecation provides the required functionality. The lookup can use only the new key, but if it does not exist and the old key exists in the configuration, it will be returned. I don't see why it is necessary to re-implement what Configuration class already provides.